### PR TITLE
Validations and model specs

### PR DESCRIPTION
**Pull request by sambeck87**
- Add the following validations:
  - For the User model:
    - Name must not be blank.
    - PostsCounter must be an integer greater than or equal to zero.
  - For the Post model:
    - Title must not be blank.
    - Title must not exceed 250 characters.
    - CommentsCounter must be an integer greater than or equal to zero.
    - LikesCounter must be an integer greater than or equal to zero.
- Add unit specs for all of your models' methods and validations.

### CODE REVIEW

File: spec/models/post_spec.rb [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    it 'post should be invalid if likes_counter = nil' do
+      subject.likes_counter = nil
+      expect(subject).to_not be_valid
+    end
+
+    it 'post should be invalid if likes_counter is negative' do
+      subject.likes_counter = -3
+      expect(subject).to_not be_valid
+    end
+  end
```
Comment (BertrandConxy) [inline spec/models/post_spec.rb:68]: - [ ] Good job so far 👏 👏 Please write tests for the custom method in your Post model which is the `recent_posts` as it is among the project requirements to do so 👇 
image

This improves confidence that your codes functions as expected 👍

File: spec/models/user_spec.rb [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+      subject.posts_counter = nil
+      expect(subject).to_not be_valid
+    end
+
+    it 'subject should be invalid if posts_counter is negative' do
+      subject.posts_counter = -3
+      expect(subject).to_not be_valid
+    end
+  end
+end
```
Comment (BertrandConxy) [inline spec/models/user_spec.rb]: - [ ] Please do the same for the User model. There is a custom method which is `recent_posts` and as it is in the project requirements, It is supposed to be tested too. Therefore, I suggest that you test it too 😃 👍

**Review by BertrandConxy — CHANGES_REQUESTED** [review]
Hi @sambeck87, 

Good job so far! 👏 👏 
There are some issues that you still need to work on to go to the next project but you are almost there! 🏃‍♂️ 🏃‍♂️ 


## Highlights 💪 
- No linter errors
- correct flow is used


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @BertrandConxy in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by tufoinnkuo10 — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi @sambeck87 

Your project is complete! Great job adding the changes 🥇. There is nothing else to say other than... it's time to merge it 🍾🚢 :shipit: . Congratulations!  🎉💯🌟 

## Good Points 👍
- Descriptive Pull Request
- model validations added
- corresponding unit test added
- No linter errors

<br>
Cheers..Happy coding!..and keep soaring higher! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 2 more limited reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**